src/serial.rs

```diff
@@ -181,7 +184,9 @@ macro_rules! hal {
                     // NOTE(unsafe) atomic read with no side effects
                     let isr = unsafe { (*$USARTX::ptr()).isr.read() };
 
-                    Err(if isr.pe().bit_is_set() {
+                    Err(if isr.rtof().bit_is_set() {
```

Comment:
I notice this error has been added, for time out detection, but I don't see any code that configures the relevant registers? Could you explain what this does and how it fixes the framing error issue?

Reply:
No actually this was an attempt to solve framing error but it seems not link to that change. I can either add the timeout support or remove this branch. Let me know. BTW I didn't saw framing error happening so far.

Reply:
As were not really sure what it does, I think its best to remove it. We can always add it back later if its required :).

Reply:
Yeap reverted